### PR TITLE
Added basic and Token-based Authentication for Remote Configuration

### PR DESCRIPTION
**Pull request by Anupam-**
Added basic and Token-based Authentication for Remote Configuration.

### CODE REVIEW

File: src/NetEscapades.Configuration.Remote/RemoteConfigurationAuthenticationTypes.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace NetEscapades.Configuration.Remote
+{
+    public enum AuthenticationTypes
```
Comment (andrewlock) [inline src/NetEscapades.Configuration.Remote/RemoteConfigurationAuthenticationTypes.cs]: Minor: Could you rename to the singular version - `AuthenticationType`, and rename the file to match please?

**Comment by andrewlock** [comment]
Thanks @Anupam- looks great 😄 

If you could just make the minor changes I'll merge it asap. 

Thanks!

**Comment by Anupam-** [comment]
Hi @andrewlock , have pushed the changes 👍. 
Please review.
Thanks

**Comment by andrewlock** [comment]
Thanks so much for the contribution, @Anupam- , appreciated! 👍

**Comment by Anupam-** [comment]
my pleasure :)

**Pull request merged**